pkg/cassandra/management.go

```diff
@@ -312,3 +331,32 @@ func (r *defaultManagementApiFacade) GetSchemaVersions() (map[string][]string, e
 
 	return nil, fmt.Errorf("failed to get schema version on all pods in CassandraDatacenter %v", utils.GetKey(r.dc))
 }
+
+func (r *defaultManagementApiFacade) HasSchemaAgreement() (bool, error) {
+	versions, err := r.GetSchemaVersions()
+	if err != nil {
+		return false, err
+	}
+
+	for uid := range versions {
+		// a key named UNREACHABLE may appear in the results when nodes are unreachable. The results
+		// from management-api will look like this:
+		//
+		//    {
+		//        "UNREACHABLE": [
+		//            "172.18.0.2"
+		//        ],
+		//        "aa573028-7c70-3b8f-a247-13bbca2011d2": [
+		//            "172.18.0.9",
+		//            "172.18.0.6"
+		//        ]
+		//    }
+		//
+		// We exclude these keys from the check.
+		if uid == "UNREACHABLE" {
+			delete(versions, uid)
```

Comment:
This is more or a question than a request for change, but is it safe to remove an element from the `versions` map while iterating over the map? I'm not as familiar with how this works in Golang, but in some other languages you may create a concurrent modification error doing this.

Reply:
This is safe operation in golang.